File: Leetcode/700-Search-In-Binary-Search-Tree/SearchInBinarySearchTree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func searchBST(root *TreeNode, val int) *TreeNode {
	cur := root
	for cur != nil {
		if cur.Val == val {
			break
		}
		if cur.Val > val {
			cur = cur.Left
			continue
		}
		cur = cur.Right
	}
	return cur
}

File: Leetcode/700-Search-In-Binary-Search-Tree/SearchInBinarySearchTree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    TreeNode* searchBST(TreeNode* root, int val) {
        TreeNode* cur = root;
        while (cur!=NULL) {
            if (cur->val == val) {
                break;
            }
            if (cur->val > val) {
                cur = cur->left;
                continue;
            }
            cur = cur->right;
        }
        return cur;
    }
};
